snowplow_tracker/test/unit/test_emitters.py

```diff
@@ -408,3 +417,61 @@ def test_input_unicode_post(self, mok_flush: Any) -> None:
 
         self.assertEqual(len(ae.buffer), 1)
         self.assertDictEqual(payload, ae.buffer[0])
+
+    @mock.patch('snowplow_tracker.Emitter.http_post')
+    def test_send_events_post_retry(self, mok_http_post: Any) -> None:
+        mok_http_post.side_effect = mocked_http_response_failure_retry
+        mok_success = mock.Mock(return_value="success mocked")
+        mok_failure = mock.Mock(return_value="failure mocked")
+
+        e = Emitter('0.0.0.0', buffer_size=10, on_success=mok_success, on_failure=mok_failure)
+        evBuffer = [{"a": "aa"}, {"b": "bb"}, {"c": "cc"}]
+        e.send_events(evBuffer)
+        
+        mok_http_post.side_effect = mocked_http_response_success
+        time.sleep(5)
+
+        mok_failure.assert_called_with(0, evBuffer)
+        mok_success.assert_called_with(evBuffer)
+
+    @mock.patch('snowplow_tracker.Emitter.http_get')
+    def test_send_events_get_retry(self, mok_http_get: Any) -> None:
+        mok_http_get.side_effect = mocked_http_response_failure_retry
+        mok_success = mock.Mock(return_value="success mocked")
+        mok_failure = mock.Mock(return_value="failure mocked")
+
+        e = Emitter('0.0.0.0', method='get', buffer_size=1, on_success=mok_success, on_failure=mok_failure)
+        evBuffer = [{"a": "aa"}, {"b": "bb"}, {"c": "cc"}]
+        e.send_events(evBuffer)
+        
+        mok_http_get.side_effect = mocked_http_response_success
+        time.sleep(5)
+
+        mok_failure.assert_called_with(0, evBuffer)
+        mok_success.assert_called_with(evBuffer)
+
+    @mock.patch('snowplow_tracker.Emitter.http_get')
+    def test_send_events_get_no_retry(self, mok_http_get: Any) -> None:
```

Comment:
I am missing the point of these last two tests – aren't they the same as the previous ones just without changing the response to success?

Also could we add a test that the custom retry status codes configuration is followed? For example by setting not to retry on status code 500 and checking that the events are not retried.

Reply:
The idea of `test_send_events_get_no_retry` was to make sure the retry isn't called with response 400 vs `test_send_events_get_retry` which fails with response 500, retries and then returns response 200 calling both `on_success` and `on_failure`.

As discussed on slack, I'll remove the custom retry codes from this PR and add it in to issue #297